File: regular_assignments/day_010/sum_of_diagonal_elements/solution.rb
class Solution
  def call
    solve([[1, -2, -3], [-4, 5, -6][-7, -8, 9]])
  end

  def solve(mat)
    r = mat.size
    c = mat[0].size

    sum = 0
    i = 0
    while i < r
      sum += mat[i][i]
      i += 1
    end
    sum
  end
end

File: regular_assignments/day_010/sum_of_diagonal_elements/solution.go
package sumOfDiagonalElements

func main() {
	solve([][]int{{1, 1, 1}, {1, 1, 1}, {1, 1, 1}})
}

func solve(matrix [][]int) int {
	sum := 0
	for i := 0; i < len(matrix); i++ {
		sum += matrix[i][i]
	}
	return sum
}
